src/test/java/io/strimzi/kafka/metrics/MetricWrapperTest.java

```diff
@@ -10,28 +10,34 @@
 import org.junit.jupiter.api.Test;
 
 import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
 public class MetricWrapperTest {
 
     @Test
     public void testLabelsFromScope() {
-        MetricWrapper mw = new MetricWrapper("", "k1.v1.k2.v2", null, "");
-        assertEquals(Labels.of("k1", "v1", "k2", "v2"), mw.labels());
-        mw = new MetricWrapper("", "k1.v1.k2.v2.", null, "");
-        assertEquals(Labels.of("k1", "v1", "k2", "v2"), mw.labels());
-        mw = new MetricWrapper("", null, null, "");
-        assertEquals(Labels.EMPTY, mw.labels());
-        mw = new MetricWrapper("", "k1", null, "");
-        assertEquals(Labels.EMPTY, mw.labels());
-        mw = new MetricWrapper("", "k1.", null, "");
-        assertEquals(Labels.EMPTY, mw.labels());
-        mw = new MetricWrapper("", "k1.v1.k", null, "");
-        assertEquals(Labels.EMPTY, mw.labels());
-
-        mw = new MetricWrapper("", "k-1.v1.k_1.v2", null, "");
-        Labels labels = mw.labels();
+        assertEquals(Labels.of("k1", "v1", "k2", "v2"), MetricWrapper.labelsFromScope("k1.v1.k2.v2", "name"));
+        assertEquals(Labels.EMPTY, MetricWrapper.labelsFromScope(null, "name"));
+        assertEquals(Labels.EMPTY, MetricWrapper.labelsFromScope("k1", "name"));
+        assertEquals(Labels.EMPTY, MetricWrapper.labelsFromScope("k1.", "name"));
+        assertEquals(Labels.EMPTY, MetricWrapper.labelsFromScope("k1.v1.k", "name"));
+
+        Labels labels = MetricWrapper.labelsFromScope("k-1.v1.k_1.v2", "name");
+        assertEquals("k_1", PrometheusNaming.sanitizeLabelName("k-1"));
+        assertEquals("v1", labels.get("k_1"));
+        assertEquals(1, labels.size());
+    }
+
+    @Test
+    public void testLabelsFromTags() {
+        Map<String, String> tags = new LinkedHashMap<>();
```

Comment:
Can we also try a regular scenario with multiple labels and no colliding names, like we have i https://github.com/strimzi/metrics-reporter/pull/38/files#diff-dc099f5ba5c7d3493db55fc8190b8badc3038f467d8edb8b97c3ca1022e3d319R22?

Reply:
Chaged to test for the golden path then reset the map to check the behavior when names collide